src/engine/renderer/Material.cpp

```diff
@@ -222,15 +222,12 @@ void UpdateSurfaceDataGeneric3D( uint32_t* materials, Material& material, drawSu
 	// u_AlphaThreshold
 	gl_genericShaderMaterial->SetUniform_AlphaTest( pStage->stateBits );
 
-	// u_InverseLightFactor
-	float inverseLightFactor = pStage->cancelOverBright ? tr.mapInverseLightFactor : 1.0f;
-	gl_genericShaderMaterial->SetUniform_InverseLightFactor( inverseLightFactor );
-
 	// u_ColorModulate
 	colorGen_t rgbGen = SetRgbGen( pStage );
 	alphaGen_t alphaGen = SetAlphaGen( pStage );
 
-	gl_genericShaderMaterial->SetUniform_ColorModulate( rgbGen, alphaGen );
+	bool mayUseVertexOverbright = pStage->type == stageType_t::ST_COLORMAP && drawSurf->bspSurface;
+	gl_genericShaderMaterial->SetUniform_ColorModulate( rgbGen, alphaGen, mayUseVertexOverbright );
```

Comment:
What is preventing to multiply `tess.svars.color` in `Tess_ComputeColor`?

Why is the material variant calling ``Tess_ComputeColor` but not the non-material one? Is that a dormant bug we missed?

Reply:
Non-material code does `Tess_ComputeColor()` in `Tess_StageIteratorColor()`. Material system does not have that.

Reply:
Ah so in non-material code, `Tess_ComputeColor()` is done before calling `Render_generic3D()`. Is it too early to multiply the light factor? Because both `tess` and `pStage` are known as well.

Reply:
> Ah so in non-material code, `Tess_ComputeColor()` is done before calling `Render_generic3D()`. Is it too early to multiply the light factor? Because both `tess` and `pStage` are known as well.

I suppose that was just to avoid having to add it to all relevant stage renderers.